community/modules/compute/gke-node-pool/variables.tf

```diff
@@ -0,0 +1,124 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+variable "project_id" {
+  description = "The project ID to host the cluster in."
+  type        = string
+}
+
+variable "cluster_id" {
+  description = "projects/{{project}}/locations/{{location}}/clusters/{{cluster}}"
+  type        = string
+}
+
+variable "name" {
+  description = "The name of the node pool. If left blank, will default to the machine type."
+  type        = string
+  default     = null
+}
+
+variable "machine_type" {
+  description = "The name of a Google Compute Engine machine type."
+  type        = string
+  default     = "c2-standard-60"
+}
+
+variable "image_type" {
+  description = "The default image type used by NAP once a new node pool is being created. Use either COS_CONTAINERD or UBUNTU_CONTAINERD."
+  type        = string
+  default     = "COS_CONTAINERD"
```

Comment:
Do you want to add some validation here? or can any image work?

Reply:
Documentation says that only these images work (in addition to a few legacy options), but my thought was to keep it flexible in case the list of options changes in the future. Without doing our own validation we see an error on terraform apply:

```
│ Error: googleapi: Error 400: Unexpected image_type: SOMETHING_CRAZY.
|
│   with module.compute_pool.google_container_node_pool.node_pool,
│   on modules/gke-node-pool-18c2/main.tf line 25, in resource "google_container_node_pool" "node_pool":
│   25: resource "google_container_node_pool" "node_pool" {
```

I am thinking this is pretty clear.